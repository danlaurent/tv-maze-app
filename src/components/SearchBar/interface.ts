export interface ISearchBar {
  search: string;
  onChangeText: (value: string) => void;
  placeholder: string;
  buttonText: string;
  onButtonPress: () => void;
  testID?: string;
}
