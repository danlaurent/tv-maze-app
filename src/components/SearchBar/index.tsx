import React from 'react';
import { View, StyleSheet } from 'react-native';
import Button from '../Button';
import SearchInput from '../SearchInput';
import { ISearchBar } from './interface';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'stretch',
  },
  inputContainer: {
    flex: 1,
    marginRight: 12,
  },
});

const SearchBar = ({
  search,
  onChangeText,
  placeholder,
  buttonText,
  onButtonPress,
  testID,
}: ISearchBar) => (
  <View testID={testID} style={styles.container}>
    <View style={styles.inputContainer}>
      <SearchInput
        value={search}
        onChangeText={onChangeText}
        placeholder={placeholder}
        testID={`${testID}_searchInput`}
      />
    </View>
    <Button
      testID={`${testID}_button`}
      text={buttonText}
      onPress={onButtonPress}
    />
  </View>
);

export default SearchBar;
