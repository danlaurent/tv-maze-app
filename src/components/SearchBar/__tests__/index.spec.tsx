import React from 'react';
import { render, RenderAPI, fireEvent } from '@testing-library/react-native';
import SearchBar from '..';

const onChangeTextMock = jest.fn();
const onButtonPressMock = jest.fn();

describe('SearchBar', () => {
  let tree: RenderAPI;

  beforeEach(() => {
    tree = render(
      <SearchBar
        search='Test Search'
        onChangeText={() => onChangeTextMock()}
        placeholder='Test Placeholder'
        onButtonPress={() => onButtonPressMock()}
        buttonText='Search'
        testID='testSearchBar'
        showButton
      />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the user types in the search bar', () => {
    beforeEach(async () => {
      await fireEvent.changeText(tree.getByTestId('testSearchBar_searchInput'));
    });

    it('fires onChangeText', () => {
      expect(onChangeTextMock).toBeCalled();
    });
  });

  describe('When the button is pressed', () => {
    beforeEach(async () => {
      await fireEvent.press(tree.getByTestId('testSearchBar_button'));
    });

    it('fires onButtonPress', () => {
      expect(onButtonPressMock).toBeCalled();
    });
  });
});
