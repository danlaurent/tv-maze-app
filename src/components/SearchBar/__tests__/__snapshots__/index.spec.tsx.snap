// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
  testID="testSearchBar"
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 12,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#121322",
          "borderColor": "#0c0d18",
          "borderRadius": 20,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 60,
          "padding": 12,
        }
      }
      testID="testSearchBar_searchInput"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Test Placeholder"
        placeholderTextColor="#6d6f7e"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "white",
            "marginLeft": 12,
          }
        }
        testID="testSearchBar_searchInput_textInput"
        underlineColorAndroid="transparent"
        value="Test Search"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 14,
        "height": 48,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
    testID="testSearchBar_button"
  >
    <Text
      style={
        Object {
          "color": "#c6c7d2",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Search
    </Text>
  </View>
</View>
`;
